File: src/main/java/bilibili/lingaly/tutorial/ModTags.java
package bilibili.lingaly.tutorial;


import net.minecraft.block.Block;
import net.minecraft.item.Item;
import net.minecraft.tag.TagKey;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class ModTags {

    public static class Blocks{

        public static TagKey<Block> valuableBlock = createBlockTagKey("valuableblock");

        public static TagKey<Block> createBlockTagKey(String name){
            return TagKey.of(Registry.BLOCK_KEY,
                    new Identifier(Tutorial.MOD_ID,
                            name));
        }

    }

    public static class Items{

        public static TagKey<Item> createItemTagKey(String name){
            return TagKey.of(Registry.ITEM_KEY,
                    new Identifier(Tutorial.MOD_ID,
                            name));
        }

    }

}

File: src/main/kotlin/bilibili/lingaly/ktmccouse/ModTags.kt
package bilibili.lingaly.ktmccouse

import net.minecraft.block.Block
import net.minecraft.item.Item
import net.minecraft.tag.TagKey
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry

object ModTags {

    object Blocks{

        val probeValuableBlock: TagKey<Block> = createBlockTagKey("probevaluableblock")

        fun createBlockTagKey(name: String): TagKey<Block>{
            return TagKey.of(
                Registry.BLOCK_KEY,
                Identifier(
                    Couse.MOD_ID,
                    name
                )
            )
        }

    }

    object Items{

        fun createItemTagKey(name: String): TagKey<Item>{
            return TagKey.of(
                Registry.ITEM_KEY,
                Identifier(
                    Couse
                        .MOD_ID,
                    name
                )
            )
        }

    }
}